src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1552,4 +1545,127 @@ public int cullInvalidConsumerContentAccess(Owner owner, String... existingModes
     }
 
+    @Transactional
+    private int postgreUnlinkIdCertificates(Collection<String> certIds) {
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer" +
+            " SET consumer_idcert_id = NULL, updated = :date" +
+            " FROM cp_consumer a, cp_id_cert c" +
+            " WHERE a.consumer_idcert_id = c.id" +
+            " AND c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    @Transactional
+    private int mariaDbUnlinkIdCertificates(Collection<String> certIds) {
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer a" +
+            " JOIN cp_id_cert c ON a.consumer_idcert_id = c.id " +
+            " SET a.consumer_idcert_id = NULL, a.updated = :date" +
+            " WHERE c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    @Transactional
+    private int postgreUnlinkCaCertificates(Collection<String> certIds) {
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer" +
+            " SET cont_acc_cert_id = NULL, updated = :date" +
+            " FROM cp_consumer a, cp_cont_access_cert c" +
+            " WHERE a.cont_acc_cert_id = c.id" +
+            " AND c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    @Transactional
+    private int mariaDbUnlinkCaCertificates(Collection<String> certIds) {
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer a" +
+            " JOIN cp_cont_access_cert c ON a.cont_acc_cert_id = c.id " +
+            " SET a.cont_acc_cert_id = NULL, a.updated = :date" +
+            " WHERE c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    /**
+     * Takes a list of identity certificate id and unlinks them from consumers.
+     *
+     * @param certIds certificate ids to be unlinked
+     * @return a number of unlinked consumers
+     * @throws PersistenceException when unsupported db dialect is detected
+     */
+    @Transactional
+    public int unlinkIdCertificates(Collection<String> certIds) throws PersistenceException {
```

Comment:
Seeing direct SQL for each backend gives me pause. What is the goal here and why can we not let Hibernate handle it nor use a different query?

Reply:
Done.